src/experimental-layers/src/solid-polygon-layer/polygon-tesselator.js

```diff
@@ -0,0 +1,262 @@
+
+  updatePositions({fp64, extruded}) {
+    const {attributes, polygons, pointCount} = this;
+
+    attributes.positions = attributes.positions || new Float32Array(pointCount * 3);
+    attributes.nextPositions = attributes.nextPositions || new Float32Array(pointCount * 3);
+
+    if (fp64) {
+      // We only need x, y component
+      attributes.positions64xyLow = attributes.positions64xyLow ||
+        new Float32Array(pointCount * 2);
+      attributes.nextPositions64xyLow = attributes.nextPositions64xyLow ||
+        new Float32Array(pointCount * 2);
+    }
+
+    updatePositions(attributes, {polygons, extruded, fp64});
+  }
+
+  indices() {
+    const {polygons, indexCount} = this;
+    return calculateIndices({polygons, indexCount});
+  }
+
+  positions() {
+    return this.attributes.positions;
+  }
+  positions64xyLow() {
+    return this.attributes.positions64xyLow;
+  }
+
+  nextPositions() {
+    return this.attributes.nextPositions;
+  }
+  nextPositions64xyLow() {
+    return this.attributes.nextPositions64xyLow;
+  }
+
+  elevations({key = 'elevations', getElevation = x => 100} = {}) {
+    const {attributes, polygons, pointCount} = this;
+    attributes[key] = attributes[key] || new Float32Array(pointCount);
+    return updateElevations(attributes[key], {polygons, getElevation});
+  }
+
+  colors({key = 'colors', getColor = x => DEFAULT_COLOR} = {}) {
+    const {attributes, polygons, pointCount} = this;
+    attributes[key] = attributes[key] || new Uint8ClampedArray(pointCount * 4);
+    return updateColors(attributes[key], {polygons, getColor});
+  }
+
+  pickingColors() {
+    return this.attributes.pickingColors;
+  }
+
+  // getAttribute({size, accessor}) {
+  //   const {polygons, pointCount} = this;
+  //   return calculateAttribute({polygons, pointCount, size, accessor});
+  // }
+}
+
+// Count number of points in a list of complex polygons
+function getPointCount(polygons) {
+  return polygons.reduce((points, polygon) => points + Polygon.getVertexCount(polygon), 0);
+}
+
+// COunt number of triangles in a list of complex polygons
+function getTriangleCount(polygons) {
+  return polygons.reduce((triangles, polygon) => triangles + Polygon.getTriangleCount(polygon), 0);
+}
+
+// Returns the offsets of each complex polygon in the combined array of all polygons
+function getPolygonOffsets(polygons) {
+  const offsets = new Array(polygons.length + 1);
+  offsets[0] = 0;
+  let offset = 0;
+  polygons.forEach((polygon, i) => {
+    offset += Polygon.getVertexCount(polygon);
+    offsets[i + 1] = offset;
+  });
+  return offsets;
+}
+
+function calculateIndices({polygons, IndexType = Uint32Array}) {
+  // Calculate length of index array (3 * number of triangles)
+  const indexCount = 3 * getTriangleCount(polygons);
+  const offsets = getPolygonOffsets(polygons);
+
+  // Allocate the attribute
+  // TODO it's not the index count but the vertex count that must be checked
+  if (IndexType === Uint16Array && indexCount > 65535) {
```

Comment:
It is not the indexCount that will overflow it is the vertexCount. The array can still have more than 64K elements, but each element can only hold 0-64K.